util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDao.java

```diff
@@ -31,23 +31,27 @@
 import org.killbill.commons.jdbi.statement.SmartFetchSize;
 import org.killbill.commons.jdbi.template.KillBillSqlDaoStringTemplate;
 import org.skife.jdbi.v2.sqlobject.Bind;
+import org.skife.jdbi.v2.sqlobject.GetGeneratedKeys;
 import org.skife.jdbi.v2.sqlobject.SqlBatch;
 import org.skife.jdbi.v2.sqlobject.SqlQuery;
 import org.skife.jdbi.v2.sqlobject.SqlUpdate;
 import org.skife.jdbi.v2.sqlobject.customizers.BatchChunkSize;
 import org.skife.jdbi.v2.sqlobject.customizers.Define;
 import org.skife.jdbi.v2.sqlobject.mixins.CloseMe;
 import org.skife.jdbi.v2.sqlobject.mixins.Transactional;
+import org.skife.jdbi.v2.util.LongMapper;
 
 @KillBillSqlDaoStringTemplate
 public interface EntitySqlDao<M extends EntityModelDao<E>, E extends Entity> extends AuditSqlDao, HistorySqlDao<M, E>, Transactional<EntitySqlDao<M, E>>, CloseMe {
 
     @SqlUpdate
+    @GetGeneratedKeys(value = LongMapper.class)
```

Comment:
Slightly tricky though: "If the columns which represent the auto-generated keys were not specified, the JDBC driver implementation will determine the columns which best represent the auto-generated keys."

Also unrelated to this specific PR, but if this is the direction to go, we should then think of [fixing our queue code](https://github.com/killbill/killbill-commons/blob/master/queue/src/main/java/org/killbill/queue/DBBackedQueue.java#L762).

Reply:
> Slightly tricky though: "If the columns which represent the auto-generated keys were not specified, the JDBC driver implementation will determine the columns which best represent the auto-generated keys."

Should be fairly safe though, as it works fine with H2/PostgreSQL/MySQL. Note that I could specify it in the annotation (`@GetGeneratedKeys(columnName = ...)`), but casing is a bit annoying (H2 would want `RECORD_ID` instead of `record_id`).

That being said, it would be nice to completely get rid of `record_id` and only rely on `id` (somewhat related: https://github.com/killbill/killbill/issues/35). This would also unblock the Spanner work (TODO mostly aspirational at this point...).

> Also unrelated to this specific PR, but if this is the direction to go, we should then think of [fixing our queue code](https://github.com/killbill/killbill-commons/blob/master/queue/src/main/java/org/killbill/queue/DBBackedQueue.java#L762).

We could potentially - one thing it keep in mind though is that `LAST_INSERT_ID()` is scoped to a connection while `RETURN_GENERATED_KEYS` is scoped to a `Statement` (would have to refresh my memory on the queue code if it matters in that case).